deploy/tools/envs-validator/schema.ts

```diff
@@ -132,12 +131,45 @@ const adButlerConfigSchema = yup
       .required(),
   });
 
+const adCustomBannerConfigSchema: yup.ObjectSchema<AdCustomBannerConfig> = yup
+  .object()
+  .shape({
+    text: yup.string(),
+    url: yup.string().test(urlTest),
+    desktopImageUrl: yup.string().test(urlTest).required(),
+    mobileImageUrl: yup.string().test(urlTest).required(),
+  });
+
+const adCustomConfigSchema = yup
```

Comment:
Have you checked the schema validation localy as it is described in [docs](https://github.com/blockscout/frontend/blob/main/docs/CONTRIBUTING.md#adding-new-env-variable)?

I guess we are missing `.json()` here, so the yup can parse the string to the actual JSON.

Also in `deploy/tools/envs-validator/index.ts` you need to replace ENV value with the content of the json-file in order to make the validator work.

```typescript
appEnvs.NEXT_PUBLIC_AD_CUSTOM_CONFIG_URL = await getExternalJsonContent(
  './public/assets/ad_custom_config.json',
  appEnvs.NEXT_PUBLIC_AD_CUSTOM_CONFIG_URL,
) || '[]';
```

Can you please temporarily add this config ENV to the `.env.main` preset file to demonstrate that everything works as expected? I will run the validator in our CI then. After that, you can comment it out. I know it is not the best workflow, but we are trying to improve it.

Reply:
@jasonzysun I have just merged some changes in the validation script, now it should be easier to add tests for the new variables. I've described all the necessary steps in the contribution guide. Check it out and let me know if you have any questions.

Reply:
That's great to hear. I haven't been able to address this section of the content lately, but I'll take care of it in the coming days.